enterprise-suite/scripts/lbc.py

```diff
@@ -284,13 +293,91 @@ def check_resource_list(cmd, expected, fail_msg):
+            claim = namespace + '/' + pvc
+
+            match = re.search(r'^([\w-]+)\s+\w+\s+\w+\s+(\w+)\s+Bound\s+('+claim+r')\s', stdout, re.MULTILINE)
+
+            if match:
+                if match.group(2) == 'Retain':
+                    list = retained
+                else:
+                    list = notRetained
+                list.append((match.group(1), match.group(3)))
+
+    return (retained, notRetained)
+
+# This is the behavior as described at
+# https://github.com/lightbend/es-backend/issues/572
+def checkPVthings(aboutToUninstall=False, namespace=None):
+    returncode, stdout, stderr = run('helm get ' + args.helm_name,
+                            DEFAULT_TIMEOUT, show_stderr=True)
+
+    if (returncode != 0) and not ('Error: release: "{}" not found'.format(args.helm_name) in stderr):
+        printerr("Unable to retrieve PV info.  Proceed with caution.")
+        return
+
+    hasPVs = 'usePersistentVolumes: true' in stdout
+
+    # This assumes the default is true.  What happens if they modify this in values.yaml?  We'll miss that I think.
+    wantsPVs = len(filter(lambda x: 'usePersistentVolumes=false' in x, sys.argv)) == 0
+
+    # debug.  remove
+    printerr("hasPVs: {}, wantsPVs: {}".format(hasPVs, wantsPVs))
+
+    if ((not hasPVs and wantsPVs) or (not hasPVs and aboutToUninstall)):
+        # This case would be typical for a dev/demo.  Chances are we're okay losing the data.
+        #    (Not sure how useful this is really.  Don't think they can (easily) grab the data.)
+        printerr("WARNING: usePersistentVolumes was false, now true. Continued (un)installation will result in the loss of Console data.")
+        fail("Stopping.  Invoke again with '--nowarn' to proceed anyway, but save your data first if so desired")
+    elif ((hasPVs and not wantsPVs) or (hasPVs and aboutToUninstall)):
+        # This case is the nasty one.  Chance of losing real data here.
+        # If    we're changing from usePersistentVolumes=true to usePersistentVolumes=false
+        #       (which means going from PV to emptyDir volume)
+        #    or
+        #       we're about to "helm delete" and usePersistentVolumes was true
+
+        if namespace == None:
+            namespace = args.namespace
+        retainedPVs, notRetainedPVs = pvsRetainedOrNot(namespace)
+
+        stop = False
+
+        if len(notRetainedPVs) > 0:
+            printerr("WARNING: Given the current and desired configs, continued (un)installation will result in the loss of Console data.")
+            for pv in notRetainedPVs:
+                printerr("   info: Reclaim policy for PV {} for claim {} is not 'Retain'".format(pv[0], pv[1]))
+            printerr("Invoke again with '--nowarn' to proceed anyway, but save your data first if so desired")
+            stop = True
+
+        if len(retainedPVs) > 0:
```

Comment:
I don't think this warning is necessary. It's also bad form to abort the install if it won't actually result in data loss. We only want to abort install as a last resort I think (e.g. if data loss will definitely happen).

Reply:
I'll change this to only spit out the warning and not abort the install.  I think it's important to let the user know about the orphaned data so they can decide what to do about it.

Reply:
sgtm